lib/qesapdeployment.pm

```diff
@@ -1814,4 +1818,117 @@ sub qesap_az_clean_old_peerings {
     }
 }
 
+=head2 qesap_az_setup_native_fencing_permissions
+
+    qesap_az_setup_native_fencing_permissions(vmname=>$vm_name,
+        subscription_id=>$subscription_id,
+        resource_group=>$resource_group,
+        role=>$role);
+
+    Sets up managed identity (MSI) by enabling system assigned identity and
+        role - role to be assigned
+        resource_group - resource group resource belongs to
+        subscription_id - valid azure subscription
+=cut
+
+sub qesap_az_setup_native_fencing_permissions {
+    my (%args) = @_;
+    my $vm_name = $args{vm_name};
+    my $subscription_id = $args{subscription_id};
+    my $resource_group = $args{resource_group};
+    my $role = 'Virtual Machine Contributor';
+
+    foreach ('vm_name', 'subscription_id', 'resource_group') {
```

Comment:
maybe better to move validation before first usage, I suggest to move it in between line 1835 and 1836. 
Then, as it is mandatory and validate, you probably do not even need to copy it to a different variable
`my $vm_name = $args{vm_name};`

```suggestion
   sub qesap_az_setup_native_fencing_permissions {
    my (%args) = @_;
     foreach ('vm_name', 'subscription_id', 'resource_group') {
        croak "Missing argument: '$_'" unless defined($args{$_});
        }
    my $vm_id = qesap_az_enable_system_assigned_identity($args{vm_name}, $args{resource_group});
    qesap_az_assign_role(assignee => $vm_id, role =>  'Virtual Machine Contributor', subscription_id =>  $args{subscription_id}, resource_group => $args{resource_group});
    }
```

Reply:
Good point. Fixed.